src/modules/rabbitmq/rabbitmq.farm.handler.service.ts

```diff
@@ -83,34 +76,65 @@ export class RabbitMQFarmHandlerService {
     async handleRewardsEvent(rawEvent: RawEventType): Promise<void> {
         const version = farmVersion(rawEvent.address);
         let event: BaseRewardsEvent;
-        let abiService: FarmAbiService;
         switch (version) {
             case FarmVersion.V1_2:
                 event = new RewardsEventV1_2(rawEvent);
-                abiService = this.abiFarmV1_2;
                 break;
             case FarmVersion.V1_3:
                 event = new RewardsEventV1_3(rawEvent);
-                abiService = this.abiFarmV1_3;
                 break;
-            case FarmVersion.V2:
-                return;
         }
 
-        const rewardPerShare = await abiService.rewardPerShare(event.address);
+        const rewardPerShare = await this.farmAbiFactory
+            .useAbi(event.address)
+            .rewardPerShare(event.address);
         const cacheKey = await this.farmSetterFactory
             .useSetter(event.address)
             .setRewardPerShare(event.getAddress(), rewardPerShare);
-        this.invalidatedKeys.push(cacheKey);
-        await this.deleteCacheKeys();
+        await this.deleteCacheKeys([cacheKey]);
 
         await this.pubSub.publish(FARM_EVENTS.CLAIM_REWARDS, {
             rewardsEvent: event,
         });
     }
 
-    private async deleteCacheKeys() {
-        await this.pubSub.publish('deleteCacheKeys', this.invalidatedKeys);
-        this.invalidatedKeys = [];
+    async handleEnterFarmEventV2(rawEvent: RawEventType): Promise<void> {
+        const event = new EnterFarmEventV2(rawEvent);
+
+        const cacheKeys = await Promise.all([
+            this.farmSetterFactory
+                .useSetter(event.address)
+                .setFarmTokenSupply(event.address, event.farmSupply.toFixed()),
+        ]);
+
+        await this.deleteCacheKeys(cacheKeys);
+    }
+
+    async handleExitFarmEventV2(rawEvent: RawEventType): Promise<void> {
+        const event = new ExitFarmEventV2(rawEvent);
+
+        const cacheKeys = await Promise.all([
+            this.farmSetterFactory
+                .useSetter(event.address)
```

Comment:
the list contains only 1 element, no need to use Promise.all here

Reply:
done